graphql/src/main/java/com/scalar/db/graphql/datafetcher/MutationDataFetcherFactory.java

```diff
@@ -0,0 +1,360 @@
+package com.scalar.db.graphql.datafetcher;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.scalar.db.api.Consistency;
+import com.scalar.db.api.Delete;
+import com.scalar.db.api.DeleteIf;
+import com.scalar.db.api.DeleteIfExists;
+import com.scalar.db.api.DistributedStorage;
+import com.scalar.db.api.DistributedTransaction;
+import com.scalar.db.api.Mutation;
+import com.scalar.db.api.Put;
+import com.scalar.db.api.PutIf;
+import com.scalar.db.api.PutIfExists;
+import com.scalar.db.api.PutIfNotExists;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.exception.transaction.CrudException;
+import com.scalar.db.graphql.schema.ScalarDbTypes;
+import com.scalar.db.graphql.schema.ScalarDbTypes.DeleteConditionType;
+import com.scalar.db.graphql.schema.ScalarDbTypes.PutConditionType;
+import com.scalar.db.graphql.schema.TableGraphQlModel;
+import com.scalar.db.io.Key;
+import graphql.Scalars;
+import graphql.execution.DataFetcherResult;
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetcherFactory;
+import graphql.schema.DataFetcherFactoryEnvironment;
+import graphql.schema.DataFetchingEnvironment;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInputObjectField;
+import graphql.schema.GraphQLScalarType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MutationDataFetcherFactory implements DataFetcherFactory<DataFetcherResult<Boolean>> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MutationDataFetcherFactory.class);
+
+  private final TableGraphQlModel tableGraphQlModel;
+  private final DistributedStorage storage;
+
+  private final DataFetcher<DataFetcherResult<Boolean>> dataFetcherForPut;
+  private final DataFetcher<DataFetcherResult<Boolean>> dataFetcherForDelete;
+  private final DataFetcher<DataFetcherResult<Boolean>> dataFetcherForBulkPut;
+  private final DataFetcher<DataFetcherResult<Boolean>> dataFetcherForBulkDelete;
+  private final DataFetcher<DataFetcherResult<Boolean>> dataFetcherForMutate;
+
+  public MutationDataFetcherFactory(
+      TableGraphQlModel tableGraphQlModel, DistributedStorage storage) {
+    this.tableGraphQlModel = tableGraphQlModel;
+    this.storage = storage;
+
+    dataFetcherForPut = this::dataFetcherForPut;
+    dataFetcherForDelete = this::dataFetcherForDelete;
+    dataFetcherForBulkPut = this::dataFetcherForBulkPut;
+    dataFetcherForBulkDelete = this::dataFetcherForBulkDelete;
+    dataFetcherForMutate = this::dataFetcherForMutate;
+  }
+
+  @Override
+  public DataFetcher<DataFetcherResult<Boolean>> get(DataFetcherFactoryEnvironment environment) {
+    GraphQLFieldDefinition fieldDefinition = environment.getFieldDefinition();
+    if (fieldDefinition.equals(tableGraphQlModel.getMutationPutField())) {
+      return dataFetcherForPut;
```

Comment:
Just a question.
It's a Java functional thing and wondering how `DataFetcherResult` is converted to  `DataFetcher<DataFetcherResult>`.
It's hard to parse only from the code.